hub/core/meta/encode/base_encoder.py

```diff
@@ -0,0 +1,179 @@
+from abc import ABC
+from typing import Any, Sequence
+from hub.constants import ENCODING_DTYPE
+import numpy as np
+
+
+LAST_SEEN_INDEX_INDEX = -1
```

Comment:
PLEASE a comment here. It is not clear why this variable is so important it needs to be factored out